### PR TITLE
Complete User Story 9 Unique Email Functionality

### LINKED ISSUE #71: User Story 9, Registration Email must be unique
```
As a visitor
When I visit the user registration page
If I fill out the registration form
But include an email address already in the system
Then I am returned to the registration page
My details are not saved and I am not logged in
The form is filled in with all previous data except the email field and password fields
I see a flash message telling me the email address is already in use
```

DO THESE STORIES FIRST:
- User Story 7, User Registration

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #74

child of #82

depends on #73

### PR DESCRIPTION
**Pull request by blake-enyart**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100%
- rspec spec/models - 100%

# What functionality does this accomplish?
Create error message specifically for when a user tries to register with email that has been used

# What did you struggle on to complete?
Boolean statement for showing just the error for the email

# Implements/Fixes:
* description
closes #71 

# Testing Changes
- [] No Tests have been changed
- [] Some Tests have been changed
- [X] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [X] My code has no unused/commented out code
- [X] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [X] I have fully tested my code

# Please include an emoji of how you feel about this branch:
👍

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -13,6 +13,12 @@ def create
     if @user.save
       flash[:alert] = "You are now registered and logged in as #{@user.email}."
       redirect_to profile_path
+    elsif @user.errors.keys.include?(:email) &&
```
Comment (milevy1) [inline app/controllers/users_controller.rb]: What if we changed line 16-18 to: `elsif User.find_by(email: @user.email)`

Comment (milevy1) [reply app/controllers/users_controller.rb]: Spoke with Blake, the intention with this was to use the errors to avoid querying the database again.
